tests/integration/wsapi/test_consumer.py

```diff
@@ -427,6 +523,172 @@ def get_job_instance_event_count():
     return models.JobInstanceEvent.objects.count()
 
 
+@database_sync_to_async
+def _prepare_activation_instance_with_vault_credentials():
```

Comment:
can you consolidate three _prepare functions into one with controlling parameters?
Similarly you may be able to combine three tests into one parameterized test.

Reply:
These are the async function tests, each test case is marked as `transaction=True`. I don't think we can separate them as  parameterized tests.